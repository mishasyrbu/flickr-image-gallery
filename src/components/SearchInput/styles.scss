.search-input {
    width: 280px;
    padding: 8px 10px;
    line-height: 20px;
    font-size: 18px;
    border-radius: 10px;
    outline: none;
    border: none;
}
