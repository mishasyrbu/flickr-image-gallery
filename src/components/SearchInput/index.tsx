import React from 'react';
import classNames from 'classnames';

import './styles.scss';

export interface Props {
    placeholder?: string,
    className?: string,
    value: string,
    onChange: Function,
}

const SearchInput: React.FC<Props> = ({ placeholder, className, value, onChange }) => {
    const handleOnChangeEvent = (event: React.SyntheticEvent) => {
        const value = (event.target as HTMLInputElement).value;
        onChange(value);
    };

    return (
        <input
            className={classNames('search-input', className)}
            type="text"
            placeholder={placeholder}
            value={value}
            onChange={handleOnChangeEvent}
        />
    );
}

export default SearchInput;
