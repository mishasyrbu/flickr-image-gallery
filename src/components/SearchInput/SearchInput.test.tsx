import React from 'react';
import { create, ReactTestRenderer } from 'react-test-renderer';

import SearchInput from './';

let snapshot: ReactTestRenderer;

beforeEach(() => {
  snapshot = create(<SearchInput value="" onChange={jest.fn()} />);
});

describe('<SearchInput />', () => {
  test('it matches the snapshot', () => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
