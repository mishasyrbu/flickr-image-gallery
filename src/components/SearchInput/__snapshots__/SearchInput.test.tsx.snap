// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> it matches the snapshot 1`] = `
<input
  className="search-input"
  onChange={[Function]}
  type="text"
  value=""
/>
`;
